packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -0,0 +1,353 @@
+        invitationHandle,
+        description,
+        customProperties,
+      );
+      return invitationP;
+    },
+    // Shutdown the entire vat and give payouts
+    shutdown: completion => {
+      E(zoeInstanceAdmin).exitAllSeats(completion);
+      dropAllReferences();
+      // @ts-ignore powers is not typed correctly: https://github.com/Agoric/agoric-sdk/issues/3239s
+      powers.exitVat(completion);
+    },
+    shutdownWithFailure,
+    assert: makeAssert(shutdownWithFailure),
+    stopAcceptingOffers: () => E(zoeInstanceAdmin).stopAcceptingOffers(),
+    makeZCFMint,
+    makeEmptySeatKit,
+
+    // The methods below are pure and have no side-effects //
+    getZoeService: () => zoeService,
+    getInvitationIssuer: () => invitationIssuer,
+    getTerms,
+    getBrandForIssuer,
+    getIssuerForBrand,
+    getAssetKind: getAssetKindByBrand,
+    /**
+     * Provide a jig object for testing purposes only.
+     *
+     * The contract code provides a callback whose return result will
+     * be made available to the test that started this contract. The
+     * supplied callback will only be called in a testing context,
+     * never in production; i.e., it is only called if `testJigSetter`
+     * was supplied.
+     *
+     * If no, \testFn\ is supplied, then an empty jig will be used.
+     * An additional `zcf` property set to the current ContractFacet
+     * will be appended to the returned jig object (overriding any
+     * provided by the `testFn`).
+     *
+     * @type {SetTestJig}
+     */
+    setTestJig: (testFn = () => ({})) => {
+      if (testJigSetter) {
+        console.warn('TEST ONLY: capturing test data', testFn);
+        testJigSetter({ ...testFn(), zcf });
+      }
+    },
+  });
+
+  // handleOfferObject gives Zoe the ability to notify ZCF when a new seat is
+  // added in offer(). ZCF responds with the exitObj and offerResult.
+  /** @type {HandleOfferObj} */
+  const handleOfferObj = Far('handleOfferObj', {
+    handleOffer: (invitationHandle, zoeSeatAdmin, seatData) => {
+      const zcfSeat = makeZCFSeat(zoeSeatAdmin, seatData);
+      const offerHandler = takeOfferHandler(invitationHandle);
+      const offerResultP = E(offerHandler)(zcfSeat).catch(reason => {
+        if (reason === undefined) {
+          const newErr = new Error(
+            `If an offerHandler throws, it must provide a reason of type Error, but the reason was undefined. Please fix the contract code to specify a reason for throwing.`,
+          );
+          throw zcfSeat.fail(newErr);
+        }
+        throw zcfSeat.fail(reason);
+      });
+      const exitObj = makeExitObj(seatData.proposal, zcfSeat);
+      /** @type {HandleOfferResult} */
+      return harden({ offerResultP, exitObj });
+    },
+  });
+
+  let contractCode;
+
+  /** @type {ZCFZygote} */
+  const zcfZygote = {
+    evaluateContract: bundle => {
+      contractCode = evalContractBundle(bundle);
+      handlePWarning(contractCode);
```

Comment:
This code is the same as before, just separated into the `evaluateContract` method.